File: quickjs/tak.js
function tak(x, y, z) {
    if (y < x) {
        return tak(
            tak(x - 1, y, z),
            tak(y - 1, z, x),
            tak(z - 1, x, y)
        );
    } else {
        return z;
    }
}

// Get arguments from the command line
const [_, x, y, z] = scriptArgs.map(Number);

console.log(tak(x, y, z));

File: clojure/tak.clj
(ns tak)

(defn tak ^long [^long x ^long y ^long z]
  (if (< y x)
    (tak
      (tak (dec x) y z)
      (tak (dec y) z x)
      (tak (dec z) x y))
    z))

(let [args *command-line-args*
      x (Long/parseLong (nth args 0))
      y (Long/parseLong (nth args 1))
      z (Long/parseLong (nth args 2))]
  (println (tak x y z)))
